src/Compilers/VisualBasic/Portable/Parser/Parser.vb

```diff
@@ -2940,37 +2940,81 @@ checkNullable:
 
         Private Function ParseTupleType(openParen As PunctuationSyntax) As TypeSyntax
             Dim elementBuilder = _pool.AllocateSeparated(Of TupleElementSyntax)()
+            Dim unexpected As GreenNode = Nothing
 
             Do
                 Dim identifierNameOpt As IdentifierNameSyntax = Nothing
                 Dim asKeywordOpt As KeywordSyntax = Nothing
 
+                ' if there is a type character or As, then this must be a name
                 If CurrentToken.Kind = SyntaxKind.IdentifierToken AndAlso
-                        PeekNextToken().Kind = SyntaxKind.AsKeyword Then
+                        (DirectCast(CurrentToken, IdentifierTokenSyntax).TypeCharacter <> TypeCharacter.None OrElse
+                        PeekNextToken().Kind = SyntaxKind.AsKeyword) Then
 
                     identifierNameOpt = ParseIdentifierNameAllowingKeyword()
```

Comment:
>ParseIdentifierNameAllowingKeyword [](start = 40, length = 34)

This logic looks inconsistent with other similar places in the language. For example, it looks like for locals and fields keywords are allowed only if there is a type character.